larray/inout/common.py

```diff
@@ -112,19 +123,40 @@ def read_items(self, keys, *args, **kwargs):
         ignore_exceptions = kwargs.pop('ignore_exceptions', False)
         self._open_for_read()
         res = OrderedDict()
-        if keys is None:
-            keys = self.list()
-        for key in keys:
+        axes_keys, groups_keys, arrays_keys = self.lists()
+        if keys is not None:
+            axes_keys = [key for key in keys if key in axes_keys]
+            groups_keys = [key for key in keys if key in groups_keys]
+            arrays_keys = [key for key in keys if key in arrays_keys]
+        # read axes
+        if len(axes_keys) > 0:
+            if display:
+                print("loading axes:", ','.join(axes_keys), "...", end=' ')
+            axes = self._read_axes(axes_keys, *args, **kwargs)
+            res.update(axes)
+            if display:
+                print("done")
+        # read groups
+        if len(groups_keys) > 0:
+            if display:
+                print("loading axes:", ','.join(groups_keys), "...", end=' ')
+            groups = self._read_groups(groups_keys, *args, **kwargs)
+            res.update(groups)
```

Comment:
a check that there is not a group overwriting an axis might be nice (but I guess its more important to do that during dump).